tcms/rpc/tests/test_testrun.py

```diff
@@ -418,6 +418,8 @@ def verify_api_with_permission(self):
         self.assertIn("product_version", result)
         self.assertIn("start_date", result)
         self.assertIn("stop_date", result)
+        self.assertIn("planned_start", result)
+        self.assertIn("planned_stop", result)
```

Comment:
This test class/scenario only validates the result but it doesn't validate the input like you have for TestRun.update method. 

Looking at NewRunForm I think there is a bug b/c the form doesn't redefined any of the date-time fields like UpdateForm does.

Add some more testing for this use-case but add it as a separate commit because I think we have to do more changes here. If I am correct (and the new test fails) I will cherry pick the first commit and you can continue working on the second one.